src/features/document/use_cases/add_document_to_path_use_case.py

```diff
@@ -3,26 +3,29 @@
 from fastapi import UploadFile
 
 from authentication.models import User
-from common.utils.string_helpers import split_dmss_ref
 from restful.request_types.shared import Entity
 from services.document_service import DocumentService
 from storage.internal.data_source_repository import get_data_source
 
 
-def add_document_to_path_use_case(
+def add_document_use_case(
     user: User,
     document: dict,
-    path_reference: str,
+    reference: str,
     files: Optional[List[UploadFile]] = None,
     update_uncontained: Optional[bool] = False,
     repository_provider=get_data_source,
 ):
     document_service = DocumentService(repository_provider=repository_provider, user=user)
     request_document: Entity = Entity(**document)
-    data_source_id, path, attribute = split_dmss_ref(path_reference)
+    if "/" not in reference:
+        # Reference only contains a data source
+        pass
+    elif "://" not in reference:
+        reference = f"/{reference}"
```

Comment:
TODO - investigate if there is a better way to handle add document to data source?